Updated CLI steps to avoid Drupal installation errors (#16)
justafish/ddev-drupal-core-dev: README.md

# ddev-core-dev

This is a DDEV addon for doing Drupal core development. 

> It assumes you already have [DDEV](https://ddev.com/) and [Docker](https://www.docker.com/products/docker-desktop/) installed, and are familiar with using `git` to clone code repos via the command line.

Reviewer: ```suggestion
It assumes you already have [DDEV](https://ddev.com/) and [a Docker provider](https://ddev.readthedocs.io/en/stable/users/install/docker-installation/) installed, and are familiar with using `git` to clone code repos via the command line.
```

```bash
git clone https://git.drupalcode.org/project/drupal.git drupal
cd drupal
ddev config --project-type=drupal10
ddev start

> ddev exec corepack enable

Reviewer: I think this should be done already by the add-on, super easy.

Author: @rfay I haven't tried running it without this line, if it's already being run I'll remove this. No point duplicating

Reviewer: No, I was saying that as a separate PR, the add-on can `corepack enable` either as a post-start hook or as a Dockerfile. I'll probably do a PR for that. I wasn't saying you should change anything here.

> ddev exec yarn set version stable

Reviewer: ```suggestion
ddev yarn set version stable
```

> ddev yarn set version stable

Reviewer: Instead, see https://github.com/justafish/ddev-drupal-core-dev/pull/17

ddev get justafish/ddev-drupal-core-dev
ddev restart

> rm -rf sites/default/files && rm -f sites/default/settings.php

Reviewer: There's no harm removing an existing .sqlite.

> rm -rf sites/default/files && rm -f sites/default/settings.php

Reviewer 2: but the .sqlite file will just be created when `ddev drupal install` is executed. at this point (in line 16) this command has no effect since there is no .sqlite file yet.

> rm -rf sites/default/files && rm -f sites/default/settings.php

Reviewer: UNLESS somebody is re-using the directory, which I often do. That's why I suggested it.

> rm -rf sites/default/files && rm -f sites/default/settings.php

Reviewer 2: hm but the readme is intended for someone starting for the first time, reading the `rm`line i consider rather confusing. but for cases like reusing the directory as well as if you are working on the installer and you have to drop the database i've added a note for on here: https://github.com/justafish/ddev-drupal-core-dev/pull/13/files in line 42

> ddev composer install

Reviewer: Might as well do it all in one step.
```suggestion
ddev composer require drush/drush && ddev composer install
```

# Install Drupal (site will be available at https://drupal.ddev.site)
ddev drupal install

> # Install Drush (recommended, use with `ddev drush <command>`)

Reviewer 2: ```suggestion
```

> # Install Drush (recommended, use with `ddev drush <command>`)
> ddev composer require drush/drush && ddev composer install

Reviewer: No longer needed
```suggestion
```

> ddev composer require drush/drush && ddev composer install

Reviewer 2: ```suggestion
```

drush shouldn't be required/installed here, that is making changes to the composer.json and you will never have a clean 11.x main branch since drush is not part of core. changes to the composer.json should only happen on a feature branch if a dependency is updated to a newer version or newly added. that is also the reason why you have the `ddev drupal cache` available from within the addon. i've added some words of caution in regards of composer require and drush in particular to my suggested changes for a PR i've created a few days ago: https://github.com/justafish/ddev-drupal-core-dev/pull/13

Reviewer 2: oh and i just noticed that `ddev composer install` was run before so the entire line could be removed imho

Author: Good point about Drush, I hadn't considered that. If the intent here is to have a totally clean codebase then it shouldn't be included, but perhaps instead a note should be added that explains why it's not present despite being such a common tool for Drupal development.
